component/azstorage/file_share.go

```diff
@@ -150,86 +154,253 @@ func (fs *FileShare) TestPipeline() error {
 }
 
 func (fs *FileShare) DeleteFile(name string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 func (fs *FileShare) DeleteDirectory(name string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 
 func (fs *FileShare) RenameFile(string, string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 func (fs *FileShare) RenameDirectory(string, string) error {
-	return nil
+	return syscall.ENOTSUP
 }
 
+// GetAttr : Retrieve attributes of a file or directory
 func (fs *FileShare) GetAttr(name string) (attr *internal.ObjAttr, err error) {
-	return nil, nil
+	log.Trace("FileShare::GetAttr : name %s", name)
+
+	// retrieve DirectoryURL of file/directory path (everything except file name)
+	// covers case where name param includes subdirectories and not just the file name
+	path := filepath.Join(fs.Config.prefixPath, name)
+	splitPath := strings.Split(path, "/")
+	splitPath = splitPath[:len(splitPath)-1]
+	joinedPath := strings.Join(splitPath, "/")
+
+	fileURL := fs.Share.NewDirectoryURL(joinedPath).NewFileURL(filepath.Base(name))
+	prop, fileerr := fileURL.GetProperties(context.Background())
+
+	if fileerr == nil { // file
+		ctime, err := time.Parse(time.RFC1123, prop.FileChangeTime())
+		if err != nil {
+			ctime = prop.LastModified()
+		}
+		crtime, err := time.Parse(time.RFC1123, prop.FileCreationTime())
+		if err != nil {
+			crtime = prop.LastModified()
+		}
+		attr = &internal.ObjAttr{
+			Path:   name, // We don't need to strip the prefixPath here since we pass the input name
+			Name:   filepath.Base(name),
+			Size:   prop.ContentLength(),
+			Mode:   0,
+			Mtime:  prop.LastModified(),
+			Atime:  prop.LastModified(),
+			Ctime:  ctime,
+			Crtime: crtime,
+			Flags:  internal.NewFileBitMap(),
+		}
+		parseMetadata(attr, prop.NewMetadata())
+		attr.Flags.Set(internal.PropFlagMetadataRetrieved)
+		attr.Flags.Set(internal.PropFlagModeDefault)
+
+		return attr, nil
+	} else if storeFileErrToErr(fileerr) == ErrFileNotFound { // directory
+		dirURL := fs.Share.NewDirectoryURL(filepath.Join(fs.Config.prefixPath, name))
+		prop, direrr := dirURL.GetProperties(context.Background())
+
+		if direrr == nil {
+			ctime, err := time.Parse(time.RFC1123, prop.FileChangeTime())
+			if err != nil {
+				ctime = prop.LastModified()
+			}
+			crtime, err := time.Parse(time.RFC1123, prop.FileCreationTime())
+			if err != nil {
+				crtime = prop.LastModified()
+			}
+			attr = &internal.ObjAttr{
+				Path:   name,
+				Name:   filepath.Base(name),
+				Size:   4096,
+				Mode:   0,
+				Mtime:  prop.LastModified(),
+				Atime:  prop.LastModified(),
+				Ctime:  ctime,
+				Crtime: crtime,
+				Flags:  internal.NewDirBitMap(),
+			}
+			parseMetadata(attr, prop.NewMetadata())
+			attr.Flags.Set(internal.PropFlagMetadataRetrieved)
+			attr.Flags.Set(internal.PropFlagModeDefault)
+
+			return attr, nil
+		}
+		return attr, syscall.ENOENT
+	}
+	// error
+	log.Err("FileShare::GetAttr : Failed to get file/directory properties for %s (%s)", name, err.Error())
```

Comment:
I think this should be
```suggestion
	log.Err("FileShare::GetAttr : Failed to get file/directory properties for %s (%s)", name, fileerr.Error())
```